src/scripts/manage_artifacts/clean_staging.py

```diff
@@ -0,0 +1,52 @@
+"""Handles cleaning the staging directory when a new release occurs"""
+import argparse
+import boto3
+
+
+def read_args() -> argparse.Namespace:
+    """Read the arguments passed to the script."""
+    parser = argparse.ArgumentParser(description="")
+    parser.add_argument("--bucket", required=True)
+    parser.add_argument("--bucket_prefix", required=True)
+    args = parser.parse_args()
+    return args
+
+
+def delete_objects(bucket_prefix: str, bucket: str) -> None:
+    """Handle cleaning up bucket in S3. This will allow the owner.txt file
+    to be kept.
+
+    Arguments:
+        bucket_prefix: The prefix of the bucket to clean. Should end with a '/'
+        bucket: The name of the bucket to clean
+    """
+    print(f"Cleaning bucket: {bucket} with prefix: {bucket_prefix}")
+
+    s3_client = boto3.client('s3')
+    response = s3_client.list_objects_v2(Bucket=bucket, Prefix=bucket_prefix)
```

Comment:
I'm thinking that we should have an exception handling here for if `bucket_prefix` is `None` or `""` (empty) because I believe `s3_client.list_objects_v2` will list all objects in the bucket if that is the case. Since `parser.add_argument("--bucket_prefix", required=True)` would handle if no `bucket_prefix` is passed perhaps we just need handling for the empty string scenario.

And because we'll be using our prod buckets, we likely want this extra safety net.

Reply:
I wonder if we should also have something here that checks that `main` is not part of the `bucket_prefix` or any of the objects listed and if it is, aborts the process.

Reply:
Thoughts on these 2 guard statements @rxu17 ?
```

    if not args.bucket_prefix or args.bucket_prefix[-1] != '/':
        raise ValueError("Bucket prefix must be provided and end with a '/'")

    if "main" in args.bucket_prefix or "main" in args.bucket:
        raise ValueError(
            "Cannot delete objects in the main directory")
```

Reply:
Looks good to me! I don't think you need the `"main" in args.bucket` but everything else works. Do you think we should have one more guard statement for the `object_key` and do something like:
```if "main" in object_key:
         raise ValueError(
            "Cannot delete objects in the main directory")
```

Reply:
Yeah that makes sense to have that extra:

> guard statement for the object_key later in that function

I will add it in